src/pages/home/report/ReportAttachments.tsx

```diff
@@ -21,12 +26,45 @@ function ReportAttachments({route}: ReportAttachmentsProps) {
     const accountID = route.params.accountID;
     const isAuthTokenRequired = route.params.isAuthTokenRequired;
     const attachmentLink = route.params.attachmentLink;
-    const [report] = useOnyx(`${ONYXKEYS.COLLECTION.REPORT}${reportID || undefined}`);
-    const [isLoadingApp] = useOnyx(ONYXKEYS.IS_LOADING_APP);
+    const [report] = useOnyx(`${ONYXKEYS.COLLECTION.REPORT}${reportID}`, {canBeMissing: true});
+    const [reportActions] = useOnyx(`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportID}`, {
+        canEvict: false,
+        canBeMissing: true,
+    });
+    const [reportMetadata] = useOnyx(`${ONYXKEYS.COLLECTION.REPORT_METADATA}${reportID}`, {
+        canBeMissing: true,
+    });
+    const [isLoadingApp] = useOnyx(ONYXKEYS.IS_LOADING_APP, {canBeMissing: true});
+    const {isOffline} = useNetwork();
     const fileName = route.params?.fileName;
 
+    const shouldFetchReport = useMemo(() => {
+        // Extract the reportActionID from the attachmentID (format: reportActionID_index)
+        const reportActionID = attachmentID?.split('_')?.[0];
+        return isEmptyObject(reportActions?.[reportActionID ?? CONST.DEFAULT_NUMBER_ID]);
+    }, [reportActions, attachmentID]);
+    const isLoading = useMemo(() => {
+        if (isOffline || isReportNotFound(report) || !reportID) {
+            return false;
+        }
+        const isEmptyReport = isEmptyObject(report);
+        return !!isLoadingApp || isEmptyReport || (reportMetadata?.isLoadingInitialReportActions !== false && shouldFetchReport);
+    }, [isOffline, reportID, isLoadingApp, report, reportMetadata, shouldFetchReport]);
+
     // In native the imported images sources are of type number. Ref: https://reactnative.dev/docs/image#imagesource
-    const source = Number(route.params.source) || route.params.source;
+    const source = Number(route.params.source) || tryResolveUrlFromApiRoot(decodeURIComponent(route.params.source));
+
+    const fetchReport = useCallback(() => {
+        openReport(reportID);
```

Comment:
Why aren't you specifying the actionId as well? It might happen that the initial items would not be present on the first page. In that case, it would still show a not found page.

Reply:
Not specifying it results in the following for chats that are very long:

https://github.com/user-attachments/assets/7d6c657b-fd65-4d49-a2d2-ab1299102211